opnode/l1/source.go

```diff
@@ -114,7 +114,7 @@ func (s Source) l1BlockRefByNumber(ctx context.Context, number *big.Int) (eth.L1
 		parentNum -= 1
 	}
 	return eth.L1BlockRef{
-		Self:   eth.BlockID{Hash: header.Hash(), Number: l1Num},
+		Self:   eth.BlockID{Hash: header.Hash(), Number: l1Num, Time: header.Time},
 		Parent: eth.BlockID{Hash: header.ParentHash, Number: parentNum},
```

Comment:
This highlights how not everywhere we can have the timestamp readily available. The parent hash, and number are known from the next block header, but time is not.

Can we please introduce `eth.TimestampedID` (or `eth.TimeBlockID` or something) so we are explicit about the timestamp requirement? I like the existing minimal ID and don't want to leave time zero for optimization, that seems prone to bugs.